public/app/plugins/datasource/jaeger/datasource.test.ts

```diff
@@ -128,14 +130,28 @@ describe('JaegerDatasource', () => {
       })
     );
     expect(mock).toHaveBeenCalledWith({
-      url: `${defaultSettings.url}/api/traces?service=jaeger-query&operation=%2Fapi%2Fservices&start=1704085200000000&end=1704106800000000&lookback=custom`,
+      url: `${defaultSettings.url}/api/traces?service=jaeger-query&operation=%2Fapi%2Fservices&${defaultSearchRangeParams}&lookback=custom`,
     });
     expect(response.data[0].meta.preferredVisualisationType).toBe('table');
     // Make sure that traceID field has data link configured
     expect(response.data[0].fields[0].config.links).toHaveLength(1);
     expect(response.data[0].fields[0].name).toBe('traceID');
   });
 
+  it("call for `query()` when `queryType === 'search'`, using default range", async () => {
+    const mock = setupFetchMock({ data: [testResponse] });
+    const ds = new JaegerDatasource(defaultSettings);
+    const query = JSON.parse(JSON.stringify(defaultQuery));
+    // @ts-ignore
+    query.range = undefined;
+
+    ds.query({ ...query, targets: [{ queryType: 'search', refId: 'a', service: 'jaeger-query', operation: ALL_OPERATIONS_KEY }] });
+
+    expect(mock).toHaveBeenCalledWith({
+      url: `${defaultSettings.url}/api/traces?service=jaeger-query&start=1704085200000000&end=1704106800000000&lookback=custom`,
+    });
+  });
+
```

Comment:
```suggestion
  it('uses default range when no range is provided for search query,', async () => {
    const mock = setupFetchMock({ data: [testResponse] });
    const ds = new JaegerDatasource(defaultSettings);
    const query = {
      ...defaultQuery,
      targets: [{ queryType: 'search', refId: 'a', service: 'jaeger-query', operation: ALL_OPERATIONS_KEY }],
      // set range to undefined to test default range
      range: undefined,
    } as unknown as DataQueryRequest<JaegerQuery>;

    ds.query(query);
    expect(mock).toHaveBeenCalledWith({
      // Check that query has time range from 6 hours ago to now (default range)
      url: `${defaultSettings.url}/api/traces?service=jaeger-query&start=1704085200000000&end=2&lookback=custom`,
    });
  });
```

We shouldn't use `@ts-ignore` and we don't need to do stringify/parse the default query. We can just:

```
    const query = {
      ...defaultQuery,
      targets: [{ queryType: 'search', refId: 'a', service: 'jaeger-query', operation: ALL_OPERATIONS_KEY }],
      // Check that query has time range from 6 hours ago to now (default range)
      range: undefined,
    } as unknown as DataQueryRequest<JaegerQuery>;
```